src/components/Class/ClassForm.js

```diff
@@ -0,0 +1,720 @@
+    return axios({
+      method: METHODS.PUT,
+      url: `${process.env.REACT_APP_MERAKI_URL}/apiDocs/classes/${classToEdit.id}`,
+      headers: {
+        accept: "application/json",
+        Authorization: user.data.token,
+        "version-code": versionCode,
+        "update-all": indicator,
+      },
+      data: payload,
+    }).then(
+      (res) => {
+        console.log("res", res);
+        //We can also change the Successfull edit class modal here.
+        //Need to change the text from create to edit
+      },
+      (error) => {
+        console.log("error", error);
+      }
+    );
+  };
+
+  const onCourseChange = (courseId) => {
+    setClassFields({ ...classFields, course_id: courseId });
+
+    if (exercisesForSelectedCourse.length > 0) {
+      return;
+    }
+    axios({
+      method: METHODS.GET,
+      url: `${process.env.REACT_APP_MERAKI_URL}/courses/${courseId}/exercises`,
+      headers: {
+        accept: "application/json",
+        "version-code": versionCode,
+        Authorization: user.data.token,
+      },
+    }).then((res) => {
+      const filteredExercises = res.data.course.exercises.filter(
+        (exercise) => exercise.content_type === "exercise"
+      );
+      setExercisesForSelectedCourse(filteredExercises);
+    });
+  };
+
+  const onExerciseChange = (exerciseId) => {
+    setClassFields({ ...classFields, exercise_id: exerciseId });
+  };
+
+  const checkForDoubtClass =
+    classFields.type === "doubt_class" &&
+    classFields.course_id !== "" &&
+    classFields.exercise_id !== "" &&
+    classFields.title !== "" &&
+    classFields.description !== "" &&
+    classFields.start_time !== "" &&
+    classFields.end_time !== ""
+      ? false
+      : true;
+
+  const submitHandle = () => {
+    const weekDday = Object.values(classFields.on_days);
+    if (classFields.type === "batch") {
+      let incrementedDate = new Date(classFields.date);
+      let onDay = incrementedDate.toString().split(" ")[0];
+      let flag = false;
+      let firstDay = "";
+      for (let i in days) {
+        if (onDay === days[i]) {
+          flag = true;
+        }
+        if (flag) {
+          for (let j of weekDday) {
+            if (days[j] === days[i]) {
+              flag = false;
+              firstDay = j;
+              setMatchDay(false);
+              break;
+            } else {
+              setMatchDay(true);
+            }
+          }
+        }
+      }
+      const index = weekDday.indexOf(firstDay);
+      if (days[firstDay] !== onDay) {
+        let newDate;
+        var i = 1;
+        while (i <= 7) {
+          incrementedDate = moment(incrementedDate).add(1, "days")._d;
+          let Day = incrementedDate.toString().split(" ")[0];
+          if (days[weekDday[index]] === Day) {
+            newDate = incrementedDate;
+            break;
+          }
+          i = i + 1;
+        }
+        classFields.date = moment.utc(newDate).format("YYYY-MM-DD");
```

Comment:
One of a number of places where I'd recommend replacing references to `moment`'s `format` with the `format` from date.js which uses `date-fns` instead of the legacy `moment`.  Note that the argument to format will changes to `yyyy-MM-dd` (as per https://github.com/date-fns/date-fns/blob/main/docs/unicodeTokens.md#popular-mistakes).  As with an earlier comment, I'll need to look more closely into these conversions between UTC and IST.

Reply:
@jschanker What should I do then? Need to change `moment`'s `format` or should I change it to `date-fns`?

Reply:
@Poonam-Singh-Bagh You'll want to use `date-fns`, but not directly.  Since you want to convert to UTC before formatting, you could use the commented out [formatInUtc](https://github.com/navgurukul/bhanwari-devi/blob/main/src/common/date.js#L235-L237) as `formatInUtc(newDate, "yyyy-MM-dd")`.  You also need to uncomment out the [formatInTimezone](https://github.com/navgurukul/bhanwari-devi/blob/main/src/common/date.js#L231-L233) and the import statement at the [top](https://github.com/navgurukul/bhanwari-devi/blob/main/src/common/date.js#L9) (https://github.com/navgurukul/bhanwari-devi/blob/main/src/common/date.js#L9)

In `moment`, `dd` will refer to the [weekday](https://momentjs.com/docs/#/displaying/format/:~:text=1st%20...%205th%206th-,dd,Su%20Mo%20...%20Fr%20Sa,-ddd).

```suggestion
        classFields.date = formatInUtc(newDate, "yyyy-MM-dd");
```